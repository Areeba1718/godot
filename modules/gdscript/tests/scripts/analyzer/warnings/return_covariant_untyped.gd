class BaseClass extends RefCounted:

	func greetings() -> void:
		print("Hello!")

class SubClass extends BaseClass:
	
	func greetings() -> void:
		print("Hello Sub Class!")

class BaseGreeting extends RefCounted:
	
	func get_obj():
		return BaseClass.new()

class SubGreeting extends BaseGreeting:
	
	func get_obj():
		return SubClass.new()

func test():
	var base:BaseGreeting = SubGreeting.new()
	var obj = base.get_obj()
	obj.greetings()
