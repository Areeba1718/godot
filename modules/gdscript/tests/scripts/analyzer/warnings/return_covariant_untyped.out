GDTEST_OK
>> WARNING
>> Line: 24
>> UNSAFE_METHOD_ACCESS
>> The method "greetings()" is not present on the inferred type "Variant" (but may be present on a subtype).
Hello Sub Class!
